test-integration/config.test.js

```diff
@@ -4,14 +4,28 @@ import readKongApi from '../src/readKongApi';
 import { configLoader } from '../src/configLoader';
 import fs from 'fs';
 import path from 'path';
+import 'core-js/features/object/from-entries';
 
 beforeEach(tearDown);
 jest.setTimeout(10000);
 
+const sortObjectProperties = obj => {
+    return Object.fromEntries(Object.entries(obj)
+        .sort(([k1, v1], [k2, v2]) => {
+            k1 > k2 ? 1 : -1;
+        })
+    );
+}
+
```

Comment:
This might work as-is right now but the key order isn't guaranteed to be stable. Here's a good explanation from SO: https://stackoverflow.com/a/5525820. It seems like we need a different diff algorithm in these tests. If we need to commit this as a stop-gap, let's at least comment this function with a warning and a link to that SO answer.

Reply:
So the answer below that one (https://stackoverflow.com/a/23202095) says the opposite: that as of ES2015, object keys are guaranteed to be returned in insertion order unless they're integers (presumably so that Arrays don't break). It seems highly unlikely that node.js will stop using insertion order in the future, and I think we don't really have to worry about kongfig running on other implementations.

That said, a different diff algorithm is probably a better way to do this. If that seems like something we'll do in the future then I would be happy to just remove this code altogether.

Reply:
TIL! I hadn't considered that ES2015 had anything to say about key order since that was also IIRC (now I'm really wondering :smile:) when `Map` was introduced. I guess `Map` goes beyond ordering keys though, dealing with the own-property issue and a few other things. I think it's still worth leaving a comment that confirms for a future reader that we're relying on ES2015 key insertion order and enumeration being the same. An alternative reading of the implementation (my first reading) was that it was attempting to clone the object and save the key order of the original object. Since in total we're also relying on the object remaining unchanged between then and the comparison, a comment will help avoid later mutations seeking to deal with other problems.

Reply:
I don't know where it was now, but I'm pretty sure I read that the property order was codified at the same time that `Map` was introduced. Not sure if that makes you more confident in your memory though 😄. 

I'm remembering now that the diffing is all done by jest's snapshot function, and I don't really want to figure out how to override that. I suppose we could do this sorting recursively, but I'm not really sure it's worth it. If anything, I'd like to reduce the size of these snapshots, and introduce other kinds of tests that are less painful to work with.